### PR TITLE
Display-name provenance for hover and diagnostics (BT-2897)

### PR DESCRIPTION
**Pull request by jamesc**
Linear issue: https://linear.app/beamtalk/issue/BT-2897

## Summary

Implements ADR 0108 Phase 4: display-name provenance so hover and diagnostics can render `RestartStrategy (#temporary | #transient | #permanent)` instead of just the bare structural expansion when a type is the eager expansion of a named `type` alias.

- Adds `TypeProvenance::Aliased { name, span }`, attached at the single point `resolve_type_annotation` eagerly expands an alias reference (`type_resolver.rs`). `InferredType::display_for_diagnostic` renders it as `AliasName (expansion)`.
- Narrowing residuals stay structural-only by design: `difference`/`intersect`/`union_of` build fresh provenance, and `union_of`'s "best provenance" heuristic explicitly excludes `Aliased` so a freshly-built union can never inherit a member's alias identity (pinned in tests, including an adversarial-review-driven regression for `A | Integer`).
- `check_impossible_singleton_comparison` now names the alias via the display layer and adds a "did you mean" suggestion for near-miss singletons against a closed union (reuses the existing `edit_distance` machinery).
- New `check_unresolved_type_aliases` validator flags annotations that closely resemble a registered alias name but don't resolve to one — reuses `DiagnosticCategory::UnresolvedClass` and is gated on cross-file metadata like the existing unresolved-class check (a strict no-op on any alias-free module).
- `compute_hover` gains an optional `AliasRegistry` parameter so hover can resolve and display alias-typed values. The real LSP `hover()` passes `None` for now — project-wide alias tracking is deferred to BT-2901 (a later ADR 0108 phase) — so this is inert for real LSP users today, exercised only by tests that build an `AliasRegistry` directly.

## Review

Went through a full 3-pass `/review-code` (diff, system, adversarial-via-subagent). Pass 1/2 caught and fixed a real gap (the unknown-alias checker wasn't gated on cross-file metadata like its sibling check — fixed). Pass 3 (adversarial) found no correctness bugs; addressed the actionable findings: tightened `tag_alias_expansion`/`alias_display_name` visibility, documented the `Aliased.span` memoization limitation and a pre-existing order-dependent `union_of` dedup edge case, and added tests for `Meta`/`Negation`/`Intersection` alias tagging and a repeated-alias-in-one-annotation regression. Filed BT-2911 for the one legitimate deferred gap: alias display doesn't yet reach the older `EcoString`-based diagnostic paths (state-field-access hover, `check_field_assignment`, `check_argument_types`).

## Test plan

- [x] `cargo test -p beamtalk-core --lib` — 5058 passed
- [x] `cargo test --workspace --all-targets` — all green except one pre-existing environmental flake (`bt2424_default_deployment_keeps_epmd_off_public_interfaces`, caused by a concurrently-running sibling agent's e2e process binding epmd system-wide on this shared dev host — unrelated to this diff, which touches only `beamtalk-core`)
- [x] `just test` (Rust + stdlib + BUnit + runtime) — all green
- [x] `just clippy`, `just fmt` — clean
- [x] `just check-corpus`, `just check-surface-drift` — clean

### CODE REVIEW

File: crates/beamtalk-core/src/semantic_analysis/type_checker/types.rs [hunk 1]
```diff
@@ -1224,7 +1411,7 @@ impl InferredType {
                 Self::make_negation(
                     (**base).clone(),
                     Self::union_of(&[(**excluded).clone(), b.clone()]),
-                    *neg_prov,
+                    neg_prov.clone(),
```
[the remark this answers is not in the document]
Comment (jamesc) [reply crates/beamtalk-core/src/semantic_analysis/type_checker/types.rs]: Confirmed and fixed in 254b194e — added `TypeProvenance::strip_alias()` and applied it at both call sites you flagged: the same-base-flattening arm in `difference` (line ~1414) and `absorb_negations`' first-negation-wins carry-over. Added a regression test (`difference_strips_alias_tag_from_the_residual_negation_arm`) mirroring the existing Union-path test but for the Negation arm, using your exact `PublicTag` repro.
